app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -202,6 +212,59 @@ class AgendaViewModel @Inject constructor(
             is OneTimeEvent.NavigateToEditEvent -> {
                 _oneTimeEvent.emit(OneTimeEvent.NavigateToEditEvent(event.eventId))
             }
+            is ShowConfirmDeleteAgendaItemDialog -> {
+                _agendaState.update {
+                    it.copy(
+                        confirmDeleteAgendaItem = event.agendaItem
+                    )
+                }
+                sendEvent(ClearErrorMessage)
```

Comment:
Same here, wouldn't it make more sense to just update the state with let's say `errorMessage = null`?

Reply:
Since its not doing an state `update` here, and just clearing the error message, i centralized it to also get rid of any progress indicator too using an event.